tensorflow_datasets/image/fruits360.py

```diff
@@ -0,0 +1,131 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Fruits-360: A dataset of images containing fruits.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from tensorflow.python.framework import errors
+
+import tensorflow_datasets.public_api as tfds
+from tensorflow import io as tfio
+
+_CITATION = """\
+@article{article,
+author = {Mureșan, Horea and Oltean, Mihai},
+year = {2018},
+month = {06},
+pages = {26-42},
+title = {Fruit recognition from images using deep learning},
+volume = {10},
+journal = {Acta Universitatis Sapientiae, Informatica},
+doi = {10.2478/ausi-2018-0002}
+}
+"""
+
+_COMMIT_SHA = 'b6960fc0287be349c222c3dbdd2624958c902431'
+_DOWNLOAD_URL = 'https://github.com/Horea94/Fruit-Images-Dataset/archive/{sha}.tar.gz'.format(sha=_COMMIT_SHA)
+
+_IMAGE_SIZE = 100
+_IMAGE_SHAPE = (_IMAGE_SIZE, _IMAGE_SIZE, 3)
+_CLASS_NAMES = ['Apple Braeburn', 'Apple Crimson Snow', 'Apple Golden 1', 'Apple Golden 2', 'Apple Golden 3',
+                'Apple Granny Smith', 'Apple Pink Lady', 'Apple Red 1', 'Apple Red 2', 'Apple Red 3',
+                'Apple Red Delicious', 'Apple Red Yellow 1', 'Apple Red Yellow 2', 'Apricot', 'Avocado', 'Avocado ripe',
+                'Banana', 'Banana Lady Finger', 'Banana Red', 'Cactus fruit', 'Cantaloupe 1', 'Cantaloupe 2',
+                'Carambula', 'Cherry 1', 'Cherry 2', 'Cherry Rainier', 'Cherry Wax Black', 'Cherry Wax Red',
+                'Cherry Wax Yellow', 'Chestnut', 'Clementine', 'Cocos', 'Dates', 'Granadilla', 'Grape Blue',
+                'Grape Pink', 'Grape White', 'Grape White 2', 'Grape White 3', 'Grape White 4', 'Grapefruit Pink',
+                'Grapefruit White', 'Guava', 'Hazelnut', 'Huckleberry', 'Kaki', 'Kiwi', 'Kohlrabi', 'Kumquats', 'Lemon',
+                'Lemon Meyer', 'Limes', 'Lychee', 'Mandarine', 'Mango', 'Mangostan', 'Maracuja', 'Melon Piel de Sapo',
+                'Mulberry', 'Nectarine', 'Orange', 'Papaya', 'Passion Fruit', 'Peach', 'Peach 2', 'Peach Flat', 'Pear',
+                'Pear Abate', 'Pear Kaiser', 'Pear Monster', 'Pear Red', 'Pear Williams', 'Pepino', 'Pepper Green',
+                'Pepper Red', 'Pepper Yellow', 'Physalis', 'Physalis with Husk', 'Pineapple', 'Pineapple Mini',
+                'Pitahaya Red', 'Plum', 'Plum 2', 'Plum 3', 'Pomegranate', 'Pomelo Sweetie', 'Quince', 'Rambutan',
+                'Raspberry', 'Redcurrant', 'Salak', 'Strawberry', 'Strawberry Wedge', 'Tamarillo', 'Tangelo',
+                'Tomato 1', 'Tomato 2', 'Tomato 3', 'Tomato 4', 'Tomato Cherry Red', 'Tomato Maroon', 'Tomato Yellow',
+                'Walnut']
+
+
+class Fruits360(tfds.core.GeneratorBasedBuilder):
+  """Fruits 360 dataset."""
+
+  VERSION = tfds.core.Version("1.0.0", experiments={tfds.core.Experiment.S3: False})
```

Comment:
New datasets should use S3 by default. The legacy mode is deprecated. Is there a reason why not using S3 ?
See here for more info: https://www.tensorflow.org/datasets/splits#two_apis_s3_and_legacy

Reply:
Sorry, didn't realise S3 was mainline since it's still in `experiments`. Will update this.